### PR TITLE
`Grading`: Fix course statistics to show correct numbers in tooltips

### PR DESCRIPTION
**Pull request by rstief**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context
Closes https://github.com/ls1intum/Artemis/issues/6323

### Description
- When hiding a score line from the "Performance in Exercises" chart, the data is now removed instead of set to 0
- This way, the tooltip does not show incorrect numbers anymore
- The data was also modified to not be off by 1 anymore (was needed before to not show the tooltip for "hidden" values)

### Steps for Testing

Prerequisites:
- A course in which the user participated in exercises

1. Log in to Artemis
2. Navigate to the Course
3. Navigate to Statistics
4. Scroll to the "Performance in Exercises" Chart
5. Hover over the chart (NOT directly over a data point)
6. You should see all three scores for an exercise
7. Toggle off a score in the legend
8. Hover over the chart again
9. You now should only see the two scores that are still toggled on

#### Performance Review
- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [exercise-scores-chart.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5390/latest/artifact/shared/Coverage-Report-Client-Tests/app/overview/visualizations/exercise-scores-chart/exercise-scores-chart.component.ts.html) | 95% | ✅ ❌ |

### Screenshots
Before the Bugfix:
image

After the Bugfix:

image
image

### CODE REVIEW

File: src/main/webapp/app/overview/visualizations/exercise-scores-chart/exercise-scores-chart.component.html [hunk 1]
```diff
@@ -46,12 +46,12 @@ <h2 class="col-9">{{ 'artemisApp.exercise-scores-chart.title' | artemisTranslate
...
             <ng-template #tooltipTemplate let-model="model">
                 <b>{{ model.name }}</b> <br />
-                <span>{{ model.series }}: {{ Math.max(model.value - 1, 0) }} %</span>
+                <span>{{ model.series }}: {{ Math.max(model.value, 0) }} %</span>
             </ng-template>
             <ng-template #seriesTooltipTemplate let-model="model">
                 <ng-container *ngIf="model.length">
                     <h6>{{ model[0].name }}</h6>
-                    <div *ngFor="let entry of model">{{ entry.series }}: {{ Math.max(entry.value - 1, 0) }}%</div>
+                    <div *ngFor="let entry of model; let i = index">{{ entry.series }}: {{ Math.max(entry.value, 0) }}%</div>
```
Comment (MaximilianAnzinger) [inline src/main/webapp/app/overview/visualizations/exercise-scores-chart/exercise-scores-chart.component.html:54]: Where do you use the index?

**Review by MaximilianAnzinger — COMMENTED** [review]
Generally looks good. I've one question.

File: src/main/webapp/app/overview/visualizations/exercise-scores-chart/exercise-scores-chart.component.html (hunk 1 above)
Comment (rstief) [reply src/main/webapp/app/overview/visualizations/exercise-scores-chart/exercise-scores-chart.component.html]: Good catch - I don't. Removed it in the latest commit.

File: src/main/webapp/app/overview/visualizations/exercise-scores-chart/exercise-scores-chart.component.ts [hunk 2]
```diff
@@ -154,30 +154,25 @@ export class ExerciseScoresChartComponent implements AfterViewInit, OnChanges {
      * @param data the event sent by the framework
      */
     onSelect(data: any): void {
```
Comment (Strohgelaender) [inline src/main/webapp/app/overview/visualizations/exercise-scores-chart/exercise-scores-chart.component.ts:156]: Can we somehow replace this `any` with an actual type?

Comment (rstief) [reply src/main/webapp/app/overview/visualizations/exercise-scores-chart/exercise-scores-chart.component.ts]: Fixed this in the most recent commit 👍

**Review by rriyaldhi — APPROVED** [review]
Tested on TS2 and works as described.

**Review by dearjasmina — APPROVED** [review]
code lgtm

**Pull request merged**